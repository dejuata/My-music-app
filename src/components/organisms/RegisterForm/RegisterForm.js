import React from 'react'
import validator from 'validator'
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom'
import { startRegisterWithEmailPasswordName } from '../../../actions/auth';
import { removeError, setError } from '../../../actions/ui';
import { useForm } from '../../../hooks/useForm'

export const RegisterForm = () => {

    const dispatch = useDispatch();
    const { msgError } = useSelector(state => state.ui)

    const [ formValues, handleInputChange ] = useForm({
        name: 'Juan David',
        email: 'demo1@gmail.com',
        password: '123456',
        password2: '123456'
    })

    const { name, email, password, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();

        if ( isFormValid() ) {
            dispatch( startRegisterWithEmailPasswordName(email, password, name) )
        }
    }

    const isFormValid = () => {

        if ( name.trim().length === 0) {
            dispatch(setError('Name is required'));
            return false;
        } else if ( !validator.isEmail( email ) ) {
            dispatch(setError('Email is not valid'));
            return false;
        } else if ( password !== password2 || password.length < 5) {
            dispatch(setError('Password should be a least 6 characters and match each other'));
            return false;
        }

        dispatch(removeError())
        return true
    }

    return (
        <form onSubmit={ handleRegister }>

            <div className='alert alert-error'>
                { msgError }
            </div>

            <div
                className='form-control mb-2'
            >
                <label
                    className='label'
                >
                    Name
                </label>

                <input
                    type='text'
                    placeholder='Your name'
                    name="name"
                    className='input'
                    autoComplete='off'
                    value={ name }
                    onChange={ handleInputChange }
                />
            </div>

            <div
                className='form-control mb-2'
            >
                <label
                    className='label'
                >
                    Email address
                </label>

                <input
                    type='email'
                    placeholder='example@email.com'
                    name="email"
                    className='input'
                    autoComplete='off'
                    value={ email }
                    onChange={ handleInputChange }
                />
            </div>

            <div
                className='form-control mb-2'
            >
                <label
                    className='label'
                >
                    Password
                </label>

                <input
                    type='password'
                    placeholder='********'
                    name="password"
                    className='input'
                    autoComplete='off'
                    value={ password }
                    onChange={ handleInputChange }
                />
            </div>

            <div
                className='form-control mb-3'
            >
                <label
                    className='label'
                >
                    Confirm Password
                </label>

                <input
                    type='password'
                    placeholder='********'
                    name="password2"
                    className='input'
                    autoComplete='off'
                    value={ password2 }
                    onChange={ handleInputChange }
                />
            </div>

            <button
                type='submit'
                className='btn btn-primary'
            >
                Sign Up
            </button>

            <Link to="/auth/login" className='auth__link'>
                Already registered ?
            </Link>
        </form>
    )
}
