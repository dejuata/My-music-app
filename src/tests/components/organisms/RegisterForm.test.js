import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from "react-redux";

import { MemoryRouter } from 'react-router-dom';
import { RegisterForm } from '../../../components/organisms/RegisterForm/RegisterForm';

const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);
// store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter>
            <RegisterForm />
        </MemoryRouter>
    </Provider>
)

describe('test RegisterForm', () => {

    beforeEach( () => {
        store = mockStore(initState);
        jest.clearAllMocks();
    })

    test('Should be render RegisterForm', () => {

        expect(wrapper).toMatchSnapshot();

    })

})