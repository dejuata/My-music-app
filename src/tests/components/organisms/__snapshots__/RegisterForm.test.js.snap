// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RegisterForm Should be render RegisterForm 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="alert alert-error"
  />
  <div
    className="form-control mb-2"
  >
    <label
      className="label"
    >
      Name
    </label>
    <input
      autoComplete="off"
      className="input"
      name="name"
      onChange={[Function]}
      placeholder="Your name"
      type="text"
      value="Juan David"
    />
  </div>
  <div
    className="form-control mb-2"
  >
    <label
      className="label"
    >
      Email address
    </label>
    <input
      autoComplete="off"
      className="input"
      name="email"
      onChange={[Function]}
      placeholder="example@email.com"
      type="email"
      value="demo1@gmail.com"
    />
  </div>
  <div
    className="form-control mb-2"
  >
    <label
      className="label"
    >
      Password
    </label>
    <input
      autoComplete="off"
      className="input"
      name="password"
      onChange={[Function]}
      placeholder="********"
      type="password"
      value="123456"
    />
  </div>
  <div
    className="form-control mb-3"
  >
    <label
      className="label"
    >
      Confirm Password
    </label>
    <input
      autoComplete="off"
      className="input"
      name="password2"
      onChange={[Function]}
      placeholder="********"
      type="password"
      value="123456"
    />
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    Sign Up
  </button>
  <a
    className="auth__link"
    href="/auth/login"
    onClick={[Function]}
  >
    Already registered ?
  </a>
</form>
`;
